.werft/platform-trigger-artificial-job.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+#
+# The easiest way to run this script is through Werft so you don't have to worry
+# about installing the appropraite service account etc. locally.
+#
+#   werft job run github -j .werft/platform-trigger-artificial-job.yaml -s .werft/platform-trigger-artificial-job.sh
+#
+
+sleep 1
+
+set -Eeuo pipefail
+
+# Activate service account and install core-dev context
+gcloud auth activate-service-account --key-file "/mnt/secrets/gcp-sa/service-account.json"
+gcloud container clusters get-credentials core-dev --zone europe-west1-b --project gitpod-core-dev
+
+echo "[Creating new branch|PHASE] Trigger new job with preview environment"
+echo "[Creating new branch] Setting up user config"
+git config --global user.name roboquat
+git config --global user.email roboquat@gitpod.io
+git remote set-url origin https://oauth2:"${ROBOQUAT_TOKEN}"@github.com/gitpod-io/gitpod.git
+
+echo "[Creating new branch] Checking out branch"
+git checkout -b "artificial-job-$(echo $RANDOM | md5sum | head -c 20; echo;)"
+echo "[Creating new branch] Creating commit"
+git commit -m "Creating an artificial traffic for the Preview Environment start SLO" --allow-empty
+echo "[Creating new branch|DONE] Trigger new job with preview environment"
+
+echo "[Trigger new job with-preview|PHASE] Trigger new job with preview environment"
+werft run github -a with-preview=true
```

Comment:
I think we need to poll for the job to finish so we can delete the branch again, otherwise we'll create a ton of branches ☺️

Reply:
This is done a few other places, e.g [here's how](https://github.com/gitpod-io/gitpod/blob/main/.werft/workspace-run-integration-tests.sh#L99) to workspace integration test jobs currently does it - we'd need something slightly different as we don't care if the preview environment failed or not, we still want to delete the branch.

Alternatively we'd need some other mechanism of deleting old branches. E.g. this job could even just start by delete all branches that match the pattern that are more than x hours old (that might be a more robust way to deal with it actually)

Reply:
> I think we need to poll for the job to finish so we can delete the branch again, otherwise we'll create a ton of branches ☺️

I am creating tons of branches but I'm never pushing them to `origin`, so it should be fine right? I don't want them to stick around anyway, so it will even get cleaned up faster if it doesn't get pushed

Reply:
Oh nice, I'm surprised this works to be honest 😝 I guess Werft doesn't validate that the Ref actually exist (e.g. [refs/heads/platform-artificial-job-7f30eab068977bb4e4d2](https://github.com/gitpod-io/gitpod/tree/refs/heads/platform-artificial-job-7f30eab068977bb4e4d2)), and it only really cares about the Revision when checking out the code ☺️ so as long as we don't use the Ref in the build job we should be good.

The only potential downside I can think of is if we decide to GC the VM because the branch doesn't exist. I can't remember if our heuristics were updated to keep VMs alive for at least 2h ☺️